### DISCUSSION TITLE
how-to-stop-visual-studio-from-creating-launchsettings-json/

### DISCUSSION THREAD

**Discussion by andrewlock** [andrewlock/blog-comments#357] [Announcements]
# how-to-stop-visual-studio-from-creating-launchsettings-json/

In this post I show how to stop Visual Studio from automatically creating launchsettings.json files for your test projects

https://andrewlock.net/how-to-stop-visual-studio-from-creating-launchsettings-json/

**Comment by vladimir-kovalyuk** [comment]
Does not seem to work with VS for Mac.

[a comment by an automated account is not shown]

**Comment by woldaker** [comment]
It would appear that it is not necessarily true that all integration tests which use a WebApplicationFactory or reference the package "Microsoft.AspNetCore.Mvc.Testing" must use the "Microsoft.NET.Sdk.Web" Sdk.

[See here for a discussion about this](https://github.com/dotnet/aspnetcore/discussions/49601#discussioncomment-6530219).

As others have noted, I also have had test projects work perfectly well without referencing the Web Sdk. I'm not sure, but it appears to me that the conditions under which this becomes true is if the app you are testing is the web app itself. If you are running an integration test which merely references and uses another project's web app via the WebApplicationFactory, I don't believe this is true.

Can anyone verify this? In the link I provided, nobody was ultimately able to answer that question of what exactly are the scenarios where using the Web Sdk becomes necessary for integration tests.
